Tests/Commands/MuseServiceTest.cpp

```
@@ -709,6 +709,90 @@ int main() {
+        // A tiny library: one wav, one decoy, one nested wav.
+        const auto libDir = std::filesystem::temp_directory_path() / "muse_service_test_lib";
+        std::filesystem::create_directories(libDir / "kicks");
+        {
+            // Real WAVs (the validator checks magic bytes, not just extension).
+            const std::string source = writeTestWav();
+            std::filesystem::copy_file(source, libDir / "top.wav",
+                                       std::filesystem::copy_options::overwrite_existing);
+            std::filesystem::copy_file(source, libDir / "kicks" / "deep.wav",
+                                       std::filesystem::copy_options::overwrite_existing);
+            std::FILE* f = std::fopen((libDir / "notes.txt").string().c_str(), "wb");
+            std::fputs("not audio", f);
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)